test/topology_custom/test_mv_backlog.py

```diff
@@ -76,3 +81,106 @@ async def test_gossip_same_backlog(manager: ManagerClient) -> None:
+                         "AND tablets = {'initial': 1}")
+    await cql.run_async(f"CREATE TABLE ks.tab (key int, c int, v text, PRIMARY KEY (key, c))")
+    await cql.run_async(f"CREATE MATERIALIZED VIEW ks.mv_cf_view AS SELECT * FROM ks.tab "
+                    "WHERE c IS NOT NULL and key IS NOT NULL PRIMARY KEY (c, key) ")
+    await wait_for_view(cql, 'mv_cf_view', node_count)
+
+    # Only remote updates hold on to memory, so make the update remote
+    srv_base = servers[0]
+    srv_view = servers[1]
+    host_base = next(h for h in hosts if h.address == srv_base.ip_addr)
+    await pin_the_only_tablet(manager, "ks", "tab", srv_base)
+    await pin_the_only_tablet(manager, "ks", "mv_cf_view", srv_view)
+
+    # All nodes in the cluster run with --smp=1, so there is only shard 0
+    shard = 0
+
+    delay_metric_name = 'scylla_storage_proxy_coordinator_mv_flow_control_delay_total'
+    throttled_writes_metric_name = 'scylla_storage_proxy_coordinator_throttled_base_writes_total'
+
+    delay_limits = [0, 500, 1000, 2000, 10000]
+    computed_delays = []
+
+    local_metrics = await manager.metrics.query(srv_base.ip_addr)
+    previous_computed_delay = local_metrics.get(delay_metric_name, shard=str(shard)) or 0.0
+    previous_total_throttled_writes = local_metrics.get(throttled_writes_metric_name, shard=str(shard)) or 0.0
+
+    stmt = cql.prepare(f"INSERT INTO ks.tab (key, c, v) VALUES (?, ?, ?)")
+
+    for delay_limit in delay_limits:
+        logger.info(f"delay_limit = {delay_limit}")
+        # Update the delay
+        await asyncio.gather(*(manager.server_update_config(s.server_id, "view_flow_control_delay_limit_in_ms", delay_limit) for s in servers))
+
+        # Make sure that view updates will hang
+        await asyncio.gather(*(manager.api.enable_injection(s.ip_addr, "never_finish_remote_view_updates", one_shot=False) for s in servers))
+
+        # Generate a large view update, it should be delayed by MV code
+        await cql.run_async(stmt, [0, 0, 100000*'a'], host=host_base)
+
+        # Measure the current delay and make sure there was exactly one throttled write,
+        # or zero if we are currently testing 0ms delay
+        local_metrics = await manager.metrics.query(srv_base.ip_addr)
+        current_computed_delay = local_metrics.get(delay_metric_name, shard=str(shard)) or 0.0
+        computed_delay = current_computed_delay - previous_computed_delay
+        current_total_throttled_writes = local_metrics.get(throttled_writes_metric_name, shard=str(shard)) or 0.0
+        if delay_limit == 0:
+            assert current_total_throttled_writes == previous_total_throttled_writes
+        else:
+            assert current_total_throttled_writes == previous_total_throttled_writes + 1
+
+        computed_delays.append(computed_delay)
+
+        previous_computed_delay = current_computed_delay
+        previous_total_throttled_writes = current_total_throttled_writes
```

Comment:
It's not clear to me why you write to `previous_*` the current value before unpausing the view update.
Wouldn't it be more reliable to fetch the `previous_*` in each iteration of the loop just before sending the write, and not in the middle of the previous iteration like you do here?

Reply:
Also there's another thing I'm wondering about. You're only doing here one big write (not a lot of small writes). How can we be confident that when the coordinator (by the way, I think we don't know which of the two nodes this is) wants to return success, it will already know the base replica (which might not be the coordinator itself!) has a high view backlog? Can't there be some delay until this base-replica statistic is known to the coordinator?

Reply:
> It's not clear to me why you write to `previous_*` the current value before unpausing the view update. Wouldn't it be more reliable to fetch the `previous_*` in each iteration of the loop just before sending the write, and not in the middle of the previous iteration like you do here?

Yeah, I get your point. It will be much more readable this way, i.e. easier for the reviewers/future readers of the test code to convince themselves that the metrics-related assertions work as intended.